zulipterminal/model.py

```diff
@@ -785,6 +786,21 @@ def get_stream_by_id(streams: List[StreamData], stream_id: int
                     sort_streams(self.pinned_streams)
                     self.controller.view.left_panel.update_stream_view()
                     self.controller.update_screen()
+        elif event['op'] == 'add':
+            def make_reduced_stream_data(stream: Any) -> StreamData:
+                # stream_id has been changed to id.
+                return StreamData({'name': stream['name'],
+                                   'id': stream['stream_id'],
+                                   'color': stream['color'],
```

Comment:
This may need simplifying to the form we use elsewhere.

Reply:
Can you explain what you mean by simplifying? I added this part from the ` _stream_info_from_subscriptions` method? The value returned was added directly to the stream dictionary after being extracted out from this function.

Reply:
I looked through the code, the other parts use this same piece of code. I picked up the pinned and unpinned streams while testing, and they use the same structure as above.
```
[
 {
   'name': 'announce', 
   'id': 1,
   'color': '#e9b', 
   'invite_only': False,
   'description': 'Stuff everyone should see. (Everyone is subscribed to this by default.)
 '},
...
 ]
```